src/main/java/org/radarcns/management/web/rest/SubjectResource.java

```diff
@@ -211,4 +228,74 @@ else if( projectId!=null) {
         subjectRepository.delete(id);
         return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(ENTITY_NAME, id.toString())).build();
     }
+
+    /**
+     * POST  /subjects/:id/sources: Assign a list of sources to the currently logged in user
+     *
+     * The request body should contain a list of sources to be assigned to the currently logged in
+     * user. If the currently authenticated user is not a subject, or not a user
+     * (e.g. client_credentials), an AccessDeniedException will be thrown. At minimum, each source
+     * should define it's device type, like so: <code>[{"deviceType": { "id": 3 }}]</code>. A
+     * source name and source ID will be automatically generated. The source ID will be a new random
+     * UUID, and the source name will be the device model, appended with a dash and the first six
+     * characters of the UUID. The sources will be created and assigned to the currently logged in
+     * user.
+     *
+     * If you need to assign existing sources, simply specify either of id, sourceId, or sourceName
+     * in the source object.
+     *
+     * @param sourceDTOS List of sources to assign
+     * @return The updated Subject information
+     */
+    @PostMapping("/subjects/{id}/sources")
+    @Timed
+    @Secured({AuthoritiesConstants.SYS_ADMIN, AuthoritiesConstants.PROJECT_ADMIN,
+              AuthoritiesConstants.PARTICIPANT})
+    public ResponseEntity<List<SourceDTO>> assignSources(@PathVariable Long id,
```

Comment:
I prefer `POST subjects/{login}/sources` for external clients, this would keep everything based on the `subject-id/login` for external partners. We don't need to additionally share the persistent layer id

Reply:
Indeed, i prefer it as well